src/ImageSharp.Drawing/Processing/Extensions/FillRectangleExtensions.cs

```diff
@@ -15,51 +15,53 @@ public static class FillRectangleExtensions
 
         /// <summary>
         /// Flood fills the image in the shape of the provided rectangle with the specified brush.
         /// </summary>
-        /// <typeparam name="TPixel">The type of the color.</typeparam>
         /// <param name="source">The image this method extends.</param>
         /// <param name="brush">The brush.</param>
         /// <param name="shape">The shape.</param>
         /// <returns>The <see cref="Image{TPixel}"/>.</returns>
-        public static IImageProcessingContext<TPixel> Fill<TPixel>(this IImageProcessingContext<TPixel> source, IBrush<TPixel> brush, RectangleF shape)
-            where TPixel : struct, IPixel<TPixel>
-            => source.Fill(brush, new RectangularPolygon(shape.X, shape.Y, shape.Width, shape.Height));
+        public static IImageProcessingContext
+            Fill(this IImageProcessingContext source, IBrush brush, RectangleF shape) =>
+            source.Fill(brush, new RectangularPolygon(shape.X, shape.Y, shape.Width, shape.Height));
 
         /// <summary>
         /// Flood fills the image in the shape of the provided rectangle with the specified brush.
         /// </summary>
-        /// <typeparam name="TPixel">The type of the color.</typeparam>
         /// <param name="source">The image this method extends.</param>
         /// <param name="options">The options.</param>
         /// <param name="color">The color.</param>
         /// <param name="shape">The shape.</param>
         /// <returns>The <see cref="Image{TPixel}"/>.</returns>
-        public static IImageProcessingContext<TPixel> Fill<TPixel>(this IImageProcessingContext<TPixel> source, GraphicsOptions options, TPixel color, RectangleF shape)
-            where TPixel : struct, IPixel<TPixel>
-            => source.Fill(options, new SolidBrush<TPixel>(color), shape);
+        public static IImageProcessingContext Fill(
+            this IImageProcessingContext source,
+            GraphicsOptions options,
+            Color color,
+            RectangleF shape) =>
+            source.Fill(options, new SolidBrush(color), shape);
 
         /// <summary>
         /// Flood fills the image in the shape of the provided rectangle with the specified brush.
         /// </summary>
-        /// <typeparam name="TPixel">The type of the color.</typeparam>
         /// <param name="source">The image this method extends.</param>
         /// <param name="color">The color.</param>
         /// <param name="shape">The shape.</param>
         /// <returns>The <see cref="Image{TPixel}"/>.</returns>
-        public static IImageProcessingContext<TPixel> Fill<TPixel>(this IImageProcessingContext<TPixel> source, TPixel color, RectangleF shape)
-            where TPixel : struct, IPixel<TPixel>
-            => source.Fill(new SolidBrush<TPixel>(color), shape);
+        public static IImageProcessingContext
+            Fill(this IImageProcessingContext source, Color color, RectangleF shape) =>
```

Comment:
Should we incorporate the overload performance fixes from https://github.com/SixLabors/ImageSharp/pull/838 here? Merging after could cause conflicts.

Reply:
Wow, did notice that. We should merge that PR ASAP, I'll have a look in the next few days.

Reply:
I'll try to create better overloads another day :)